server/eclipse-project/src/main/java/us/freeandfair/corla/query/CastVoteRecordQueries.java

```diff
@@ -0,0 +1,266 @@
+ */
+public final class CastVoteRecordQueries {
+  /**
+   * The "county ID" field.
+   */
+  private static final String COUNTY_ID = "my_county_id";
+  
+  /**
+   * The "imprinted ID" field.
+   */
+  private static final String IMPRINTED_ID = "my_imprinted_id";
+  
+  /**
+   * The "record type" field.
+   */
+  private static final String RECORD_TYPE = "my_record_type";
+  
+  /**
+   * The "timestamp" field.
+   */
+  private static final String TIMESTAMP = "my_timestamp";
+  
+  /**
+   * The "could not query database for CVRs error message.
+   */
+  private static final String COULD_NOT_QUERY_DATABASE = 
+      "could not query database for CVRs";
+  
+  /**
+   * Private constructor to prevent instantiation.
+   */
+  private CastVoteRecordQueries() {
+    // do nothing
+  }
+  
+  /**
+   * Obtain a stream of CastVoteRecord objects with the specified type, 
+   * ordered by their imprinted ID. This method <em>must</em>
+   * be called from within a transaction, and the result stream must be
+   * used within the same transaction.
+   *
+   * @param the_type The type.
+   * @return the stream of CastVoteRecord objects, or null if one could
+   * not be acquired.
+   * @exception IllegalStateException if this method is called outside a 
+   * transaction.
+   */
+  public static Stream<CastVoteRecord> getMatching(final RecordType the_type) {
+    if (!Persistence.isTransactionRunning()) {
+      throw new IllegalStateException("no running transaction");
+    }
+    
+    Stream<CastVoteRecord> result = null;
+   
+    try {
+      final Session s = Persistence.currentSession();
+      final CriteriaBuilder cb = s.getCriteriaBuilder();
+      final CriteriaQuery<CastVoteRecord> cq = cb.createQuery(CastVoteRecord.class);
+      final Root<CastVoteRecord> root = cq.from(CastVoteRecord.class);
+      cq.select(root).where(cb.equal(root.get(RECORD_TYPE), the_type));
+      cq.orderBy(cb.asc(root.get(IMPRINTED_ID)));
+      final Query<CastVoteRecord> query = s.createQuery(cq);
+      result = ((Query<CastVoteRecord>) query).stream();
+    } catch (final PersistenceException e) {
+      Main.LOGGER.error(COULD_NOT_QUERY_DATABASE);
+    }
+    if (result == null) {
+      Main.LOGGER.info("found no CVRs for type " + the_type);
+    } else {
+      Main.LOGGER.info("query succeeded, returning CVR stream");
+    }
+    return result;
+  }
+
+  /**
+   * Obtain a stream of CastVoteRecord objects with the specified county 
+   * and type, ordered by their imprinted ID. This method <em>must</em>
+   * be called from within a transaction, and the result stream must be
+   * used within the same transaction.
+   *
+   * @param the_county The county.
+   * @param the_type The type.
+   * @return the stream of CastVoteRecord objects, or null if one could
+   * not be acquired.
+   * @exception IllegalStateException if this method is called outside a 
+   * transaction.
+   */
+  public static Stream<CastVoteRecord> getMatching(final Integer the_county,
```

Comment:
I'll have to think about that. I'm not used to thinking in streams and Java at the same time, this just seemed like a better idea than trying to use a JDBC `ScrollableResultSet` given its availability.